clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -872,6 +875,10 @@ public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
     public void unsubscribe() {
         acquire();
         try {
+            // make sure the offsets of topic partitions the consumer is unsubscribing from
+            // are committed since there will be no following rebalance
+            this.coordinator.maybeAutoCommitOffsetsAsync(Long.MAX_VALUE);
```

Comment:
Aha, I missed the significance of the parameter. We actually want the commit to happen regardless of the next auto commit deadline. So maybe we can refactor as you suggested and have the public version accept no arguments and simply send the offset commit. Maybe we should call it `maybeAutoCommitNow` or something like that? Then the existing method can delegate to this method when the commit deadline has been reached.